pylearn2/datasets/tests/test_preprocessing.py

```diff
@@ -194,66 +194,124 @@ def test_rgb_yuv():
-        abs_difference = np.abs(identity - matrix)
-        return (abs_difference < .0001).all()
-
-    assert preprocessor.P_.shape == (X.shape[1], X.shape[1])
-    assert not is_identity(preprocessor.P_)
-    assert is_identity(np.dot(preprocessor.P_, preprocessor.inv_P_))
+class testZCA:
 
+    def setup(self):
+        """
+        We use a small predefined 8x5 matrix for
+        which we know the ZCA transform.
+        """
+        self.X = np.array([[-10.0, 3.0, 19.0, 9.0, -15.0],
+                          [7.0, 26.0, 26.0, 26.0, -3.0],
+                          [17.0, -17.0, -37.0, -36.0, -11.0],
+                          [19.0, 15.0, -2.0, 5.0, 9.0],
+                          [-3.0, -8.0, -35.0, -25.0, -8.0],
+                          [-18.0, 3.0, 4.0, 15.0, 14.0],
+                          [5.0, -4.0, -5.0, -7.0, -11.0],
+                          [23.0, 22.0, 15.0, 20.0, 12.0]])
+        self.dataset = DenseDesignMatrix(X=as_floatX(self.X),
+                                         y=as_floatX(np.ones((8, 1))))
+        self.num_components = self.dataset.get_design_matrix().shape[1] - 1
 
-def test_zca_inverse():
-    """
-    Calculates the inverse of X with numpy.linalg.inv if inv_P_ is not stored
-    """
-
-    def test(store_inverse):
-        rng = np.random.RandomState([1, 2, 3])
-        X = as_floatX(rng.randn(15, 10))
-        preprocessed_X = copy.copy(X)
-        preprocessor = ZCA(store_inverse=store_inverse)
+    def is_almost_equal(self, A, B, epsilon=1e-4):
+        abs_difference = np.abs(A - B)
+        return (abs_difference < epsilon).all()
 
-        dataset = DenseDesignMatrix(X=preprocessed_X,
+    def get_preprocessed_data(self, preprocessor):
+        X = copy.copy(self.X)
+        dataset = DenseDesignMatrix(X=X,
                                     preprocessor=preprocessor,
                                     fit_preprocessor=True)
+        return dataset.get_design_matrix()
 
-        preprocessed_X = dataset.get_design_matrix()
+    def test_zca(self):
+        """
+        Confirm that ZCA.inv_P_ is the correct inverse of ZCA.P_.
+        There's a lot else about the ZCA class that could be tested here.
+        """
+        preprocessor = ZCA()
+        preprocessor.fit(self.X)
+
+        identity = np.identity(self.X.shape[1], theano.config.floatX)
+        # Check some basics of transformation matrix
+        assert preprocessor.P_.shape == (self.X.shape[1], self.X.shape[1])
+        assert self.is_almost_equal(np.dot(preprocessor.P_,
+                                           preprocessor.inv_P_), identity)
+
+        preprocessor = ZCA(filter_bias=0.0)
+        preprocessed_X = self.get_preprocessed_data(preprocessor)
+
+        #Check if preprocessed data matrix is white
+        assert self.is_almost_equal(np.cov(preprocessed_X.transpose(),
+                                           bias=1), identity)
+
+        #Check if we obtain correct solution
```

Comment:
Lines 244 and 248 here are causing the build to fail (style guide says "#" has to have a space after it.)